nvtabular/workflow.py

```diff
@@ -853,37 +681,113 @@ def exec_phase(self, phase_index, record_stats=True):
 
         # Compute statistics if necessary
         if stats:
-            for r in self.client.compute(stats):
-                computed_stats, op = r.result()
-                op.dask_fin(computed_stats)
-                self._update_stats(op)
-                op.clear()
+            if self.client:
+                for r in self.client.compute(stats):
+                    computed_stats, op = r.result()
+                    op.dask_fin(computed_stats)
+                    self._update_stats(op)
+                    op.clear()
+            else:
+                for r in dask.compute(stats, scheduler="synchronous")[0]:
+                    computed_stats, op = r
+                    op.dask_fin(computed_stats)
+                    self._update_stats(op)
+                    op.clear()
             del stats
 
     def apply(
         self,
         dataset,
         apply_offline=True,
-        record_stats=True,
+        record_stats=None,
```

Comment:
The docstring says that record_stats is a boolean (and it seems to be used that way). Should the default value by False ?
```suggestion
        record_stats=False,
```

Reply:
We probably want the default to be True, but we then we cannot raise a warning when `record_stats=True` is used with `apply_offline=True`. (we just won't calculate statistics online ever...) -  My thought with using `None` was that the default can depend on the value of `apply_offline`